src/lib.rs

```diff
@@ -181,6 +181,88 @@ pub enum Attr {
     BackgroundColor(color::Color)
 }
 
+/// An error arising from interacting with the terminal.
+#[derive(Debug)]
+pub enum Error {
+    /// Indicates an error from any underlying IO
+    Io(io::Error),
+    /// Indicates an error during terminfo parsing
+    TerminfoParsing(terminfo::Error),
+    /// Indicates an error expanding a parameterized string from the terminfo database
+    ParameterizedExpansion(terminfo::parm::Error),
+    /// Indicates that the terminal does not support the requested attribute
+    AttributeNotSupported,
```

Comment:
Can we change this to a more generic `NotSupported` (or `OperationNotSupported`)? #50 needs a way to say that querying the terminal size isn't supported.